CONTRIBUTING.md

```diff
@@ -0,0 +1,115 @@
+
+This project only uses *publicly available code*. In particular, do not use any Nintendo source code leaks.
+
+### Environment Setup
+
+Get started by following our [installation guide](https://github.com/zeldaret/mm/wiki/Installation).
+When successful, you should be able to build a matching ROM before you start making any changes.
+
+### First Contribution
+
+Usually, the best place to get started is to decompile an actor overlay. 
+An *actor* is any thing in the game that moves or performs actions or interactions: Link is an actor, enemies are actors, NPCs are actors, etc.
+
+You can [join the Discord](https://discord.zelda64.dev/) to say hello and get suggestions on where to start on the `#mm-decomp` channel.
+We track who is working on what on the [Trello board](https://trello.com/b/ruxw9n6m/majoras-mask-decompilation).
+
+Style Guide & Conventions
+-------------------------
+
+Most of the C formatting style is enforced by the `format.sh` script, which is based on `clang-format`.
+Running `./format.sh` will apply our standard style to all `.c` files in the repository.
+
+There are some conventions that cannot be automatically enforced.
+
+(TODO)
+- Naming scheme
+- `ActorFunc` naming specifics
+- Renaming symbols referenced by asm (?)
+- Check for new warnings
+
+### `NON_MATCHING`
+
+If you make significant progress on decompiling a function, but can't get it to match perfectly, you can use a `NON_MATCHING` block to commit your code but have it excluded from the build, like so:
+
+```c
+#ifdef NON_MATCHING
+void CollisionCheck_SpawnWaterDroplets(GlobalContext* ctxt, Vec3f* v) {
+    // ... 
+}
+#else
+#pragma GLOBAL_ASM("./asm/non_matchings/code/z_collision_check/CollisionCheck_SpawnWaterDroplets.asm")
+#endif
+```
+
+Pull Requests (PRs)
+-------------------
+
+### Checklist
+
+Before opening a PR, walk through the following steps to ensure that your code confroms to the style guide & conventions.
+
+- Ensure `make` successfully builds a matching ROM.
+- Run `./format.sh` to reformat your updated files.
+- The MIPS C compiler should not produce additional warnings. 
+    - In most cases, you can check this by `touch`ing (or re-saving) your modified `.c` files, and re-running `make`.
+- New variables & functions should follow the naming conventions above.
+- Check spelling in comments & renamed symbols
+
+TODO: Do we care about commit messages, hygine, etc?
+
+### Pull Request Process
+
+TODO: What should go in the pull request body?
+
+After opening a PR, the Jenkins server will verify your changes. If there is an error, double-check that you can successfully `make clean && make` locally and that all added/modified files were `git add`-ed to your commit.
+
+Each PR needs a review from one reviewer, plus the project lead. The project lead can then merge the code once all comments are addressed. (TODO: add more detail here)
```

Comment:
Oh other question: who marks comments as "resolved"? The person who started the thread, or the code author? 

(I prefer having the person who started the thread mark it as resolved; but I don't care as long as we can make it explicit here)

Reply:
Well, I can't mark as resolved conversations I have started personally. I think it may be a misconfiguration of the repo.
For now, it is fine if the author of the PR marks them as resolved.

Reply:
I think we need to have a discussion about this on discord about this. I see positives on both sides of the argument

Reply:
https://discord.com/channels/688807550715560050/740615251011174401/844345330911674368
We discussed this on Discord.
Because of the permissions structure of the repository, we will go with the following setup:
The PR author marks the comments as resolved, since they are the ones with permissions. Once all comments are addressed, it is courteous to ping the reviewer on either Discord or GitHub via requesting a review.

Before a merge, an approval from both a contributor and dev review are required. It will also be up to dev lead review to ensure that all procedures are followed.